react/src/Header.js

```diff
@@ -0,0 +1,258 @@
+import './css/App.css';
+import React from 'react';
+import { useContext } from 'react';
+import { Link, NavLink, useNavigate } from 'react-router-dom';
+import HeaderData from './HeaderData';
+import { BarSpinner, StandardSpinner } from './Spinners';
+import Auth from './utils/Auth';
+import Client from './utils/Client';
+import Context from './utils/Context';
+import LiveTime from './LiveTime';
+import Env from './utils/Env';
+import Image from './utils/Image';
+import Logout from './utils/Logout';
+import { ReadOnlyModeDisplay } from './ReadOnlyMode';
+import { useFetching } from './utils/Fetch';
+// import JustLoggedIn from './JustLoggedIn';
+// Issues with serving images ONLY from 4dn-dcic/dev NOT from cgap-supertest ...
+// So serve from my GitHub account for now ...
+// import LockImage from './media/lock.jpg';
+// import UnlockImage from './media/unlock.jpg';
+
+const Header = (props) => {
+
+    const [ header ] = useContext(HeaderData);
+    //
+    // Very odd but this below (navigate) declartion of useNavigate is REQUIRED, even if not
+    // used here, in order for the header navigation links (e.g. HOME, INFO) to work properly.
+    // If this is not here, for example, then the (target) values do not get updated properly
+    // when on the EnvPage and clicking from one environment to another. No idea why.
+    // Fun tracking this down.
+    //
+    const navigate = useNavigate();
+    const [ fetching ] = useFetching();
+
+    let titleBackgroundColor = Env.IsFoursightFourfront(header) ? "#14533C" : "#143C53";
+    let subTitleBackgroundColor = Env.IsFoursightFourfront(header) ? "#AEF1D6" : "#AED6F1";
```

Comment:
All this CSS/styles related stuff needs to be majorly refactored. (I did find/try a way to nicely globally change styles based on some dynamic value, e.g. foursight cgap vs fourfront, so that will be doable, but of course requires reworking these usages).